### ISSUE TITLE
Not working for ios 11

### ISSUE THREAD

**Issue by havenht** [silkimen/cordova-plugin-advanced-http#51]
I use cordova.plugin.http.post and it keep go to error response. ios 10 and below work fine.

**Comment by ganey** [comment]
Also having this issue :(

**Comment by silkimen** [comment]
Hi guys,

can you please provide more info?

- cordova version
- cordova iOS platform version
- plugin version
- which data serializer are you using?

**Comment by ganey** [comment]
Apologies, 

I was on Cordova 7.0.1, cordova-ios 4.5.3, I can't remember the others though.

I wasn't specifying a serializer, does that make it default to urlencoded? 

I have since re-setup my project and reinstalled the plugins and node modules and everything is working correctly.

I've been using it with Ionic, and it's possible that it was just an issue with how I was using the plugin, rather than a genuine issue. The API server I am working with can fail with a 404 on the options request if data is passed incorrectly (not the best response code I know). So I think blindly updating plugins without checking changes and not consulting your docs has been the issue with the project I'm working on. 

**TLDR**: Reinstalling the plugin and node modules has fixed the issue. I have been unable to replicate since.

**Comment by havenht** [comment]
cli-7.0.1 , ios: 4.4.0 , cordova-plugin-advanced-http   version: 1.9.0, adobe phonegap build site,

**Comment by silkimen** [comment]
@ganey thanks for your feedback!

@havenht can you show us some example code and result (error message)? You can try to post some data to http://httpbin.org/post for example.

**Comment by havenht** [comment]
i use your basic examples for post and get. Both doesn't work for ios 11, but work for ios 10 and below. You should test it out on ios 11 on your end.

cordova.plugin.http.get('https://google.com/', {
  id: 12,
  message: 'test'
}, { Authorization: 'OAuth2: token' }, function(response) {
  console.log(response.status);
}, function(response) {
  console.error(response.error);
});

**Comment by silkimen** [comment]
Actually, I did already test on iOS 11 (Simulator and Device) and I don't face any problems.

Atm, my specs are configured to run against iOS 11 locally on my dev machine (see [here)](https://github.com/silkimen/cordova-plugin-advanced-http/blob/master/test/app-mocha-specs/helpers/caps.js#L12-L18) and to run against iOS 10.3 on saucelabs (see [here](https://github.com/silkimen/cordova-plugin-advanced-http/blob/master/test/app-mocha-specs/helpers/caps.js#L42-L48)).

You can have a look into [this file](https://github.com/silkimen/cordova-plugin-advanced-http/blob/master/test/app-test-definitions.js#L132-L138) to see the specs I've implemented. [Here](https://travis-ci.org/silkimen/cordova-plugin-advanced-http) you can see the result logs of my commits built on Travis and tested on SauceLabs.

**Comment by nikmartin** [comment]
I am getting the http 200 response in the error handler as well. I'm using the ionic native wrapper if that matters

**Comment by silkimen** [comment]
Hi nikmartin,

is your backend a public webservice I can reach and test against? Did you already try to post some data to http://httpbin.org/post? I'd like to find out if it's caused by a specific value in the response, because I don't face any of the mentioned problems on my side.

**Comment by nikmartin** [comment]
> is your backend a public webservice I can reach and test against?

Yes, it's a shared iCal file located here: https://outlook.office365.com/owa/calendar/f601ad847ba74f0b87dc64989512bd4d@mcpss.com/46c96d0559e749a796af4aca6623c02d15003821461964994350/calendar.ics

I fetch it using ( this is angular/typescript using the http native ionic plugin wrapper): 
 
```
public getCalendar(url): Promise<any> {
    return new Promise((resolve, reject) => {
        this.http.setRequestTimeout(10);
        this.http.get(url, {}, {}).then(
          icalResp => {
            console.log(icalResp);
            this._rawICAL = icalResp.data;
            
            resolve();
          }).catch(error => {
            console.log(error);
          }
        );
    });
  }
```

And the catch receives a 200 status and the message is the actual output

**Comment by nikmartin** [comment]
> Did you already try to post some data to http://httpbin.org/post?

I will try this now and let you know. BTW the plugin works fine with that URL on Android, this is only on iOS, but occurs on every version I've tried, from 9.3 to current.

**Comment by nikmartin** [comment]
POST-ing from the app to httpbin/org/post using the plugin on iOS works. Here's the response when I run the function above, with the iCal url:

```
{"status":200,"headers":{"x-beserver":"DM5PR08MB2922","set-cookie":"ClientId=111A2D119B5544509C377D635DD57850; expires=Thu, 03-Jan-2019 20:46:01 GMT; path=/; secure, OutlookSession=98100911c1de429dad86ff6dd226eacb; path=/; secure; HttpOnly","content-encoding":"gzip","server":"Microsoft-IIS/10.0","x-calculatedbetarget":"DM5PR08MB2922.namprd08.prod.outlook.com","cache-control":"private","strict-transport-security":"max-age=31536000; includeSubDomains, max-age=31536000; includeSubDomains","x-ua-compatible":"IE=EmulateIE7","x-powered-by":"ASP.NET","x-calculatedfetarget":"DM5PR2201CU002.internal.outlook.com","x-feproxyinfo":"DM5PR2201CA0052.NAMPRD22.PROD.OUTLOOK.COM","request-id":"36be9174-2547-4673-aa62-cbe232e976f9","content-length":"6101","vary":"Accept-Encoding","date":"Wed, 03 Jan 2018 20:46:01 GMT","x-diaginfo":"DM5PR08MB2922","x-feserver":"DM5PR2201CA0052, BN6PR2201CA0034","x-backendhttpstatus":"200, 200","content-type":"text/calendar; charset=utf-8","x-content-type-options":"nosniff","datasource":"Exo"},"url":"https://outlook.office365.com/owa/calendar/564ead107010487ca5dcc420d636212b@mcpss.com/23302316d9e143efab2c5445061f2d1815524450471542560085/calendar.ics","error":"BEGIN:VCALENDAR\r\nMETHOD:PUBLISH\r\nPRODID:Microsoft Exchange Server 2010\r\nVERSION:2.0\r\nX-WR-CALNAME:Dawes Calendar 2018\r\nBEGIN:VTIMEZONE\r\nTZID:Central Standard Time\r\nBEGIN:STANDARD\r\nDTSTART:16010101T020000\r\nTZOFFSETFROM:-0500\r\nTZOFFSETTO:-0600\r\nRRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=1SU;BYMONTH=11\r\nEND:STANDARD\r\nBEGIN:DAYLIGHT\r\nDTSTART:16010101T020000\r\nTZOFFSETFROM:-0600\r\nTZOFFSETTO:-0500\r\nRRULE:F
....
}
```

**Comment by nikmartin** [comment]
This may be of interest:

` "content-type":"text/calendar; charset=utf-8" `

**Comment by nikmartin** [comment]
I temporarily am working around the issue by just checking the errror status code and if it's 200 assume that 'error' is actually 'data'.

If you would like me to debug anything further, I'll be glad to. If it was Android I would have already done a pull request with a patch, but I'm not an iOS dev by any stretch of the imagination.

**Comment by silkimen** [comment]
Ah ok, now I got it! 💡

This issue is caused by the content type header "text/calendar". There is a white-list of allowed content types ([see here](https://github.com/silkimen/cordova-plugin-advanced-http/blob/master/src/ios/TextResponseSerializer.m#L26)) and "text/calendar" is not listed there.

I think the white-list was meant to prevent problems with not supported data types, but it's only causing problems. Should be removed. I'll create a new issue referencing this one.

**Comment by silkimen** [comment]
I'm closing this issue now. Please feel free to re-open it, if #58 is not fixing your problem.
